### PR TITLE
Implements dummy tagging handlers for Terraform

### LINKED ISSUE #6310: Terraform AWS provider ERROR: NotImplemented: A header you provided implies functionality that is not implemented
## Expected behaviour
We're using terraform with AWS provider (version 1.32) to interact with Minio (aws_s3_bucket_object) 
It works perfectly fine with any version below **RELEASE.2018-05-04T23-13-12Z**

## Actual behaviour
We're using terraform with AWS provider (version 1.32) to interact with Minio (aws_s3_bucket_object) 
It works perfectly fine until version: **RELEASE.2018-05-04T23-13-12Z** where provider reports the below error:

```
aws_s3_bucket_object.ssl-cert[3]: aws_s3_bucket_object.ssl-cert.3: Failed to get object tags (bucket: /my-bucket/certs, key: cert.key): NotImplemented: A header you provided implies functionality that is not implemented
	status code: 501, request id: 154BB73EA25EC680, host id: 
```

## Steps to reproduce the behavior
Use below terraform code to reproduce the issue:

```
provider "aws" {
    skip_requesting_account_id = true
    skip_credentials_validation = true
    skip_get_ec2_platforms = true
    skip_metadata_api_check = true       
    access_key = "${var.access_key}"
    secret_key = "${var.secret_key}"
    region     = "us-east-1"
    endpoints {
        s3     = "${var.s3_endpoint}"
  }  
}

resource "aws_s3_bucket_object" "ssl-cert" {
    bucket = "/my-bucket/certs}"
    key    = "${var.ssl-file}"
    source = "generated/${var.ssl-file}"
    content_type = "text/plain"
    etag   = "${md5(file("generated/${var.ssl-files}"))}"
}
```

## mc version
We're not using mc. 

## System information
Terraform 0.11.7
Terraform AWS provider (version 1.32)

### PR DESCRIPTION
**Pull request by ebozduman**
Fixes #6310 

## Description
Adds dummy handlers for those not implemented functions.

## Motivation and Context
Bug fix

## Regression
No, it is not a regression.

## How Has This Been Tested?

This setup tests if a bucket can be created with terraform.

 - Install Terraform, setup your path etc.
 - Run `terrarform init`
 - Start minio server (secure - HTTPS)
 - Create a minio.tf script with the following information under Terraform installation directory:
```
provider "aws" {
  skip_requesting_account_id = true
  skip_credentials_validation = true
  skip_get_ec2_platforms = true
  skip_metadata_api_check = true       
  s3_force_path_style = true
  access_key = "minio"
  secret_key = "minio123"
  region     = "us-east-1"
  endpoints {
        s3   = "192.168.1.124:9000"
  }

}

resource "aws_s3_bucket" "bucket" {
  bucket = "abc"
  acl    = "public"
  region     = "us-east-1"
}
```
 - Run `terraform apply`

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added unit tests to cover my changes.
- [ ] I have added/updated functional tests in [mint](https://github.com/minio/mint). (If yes, add `mint` PR # here: )
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-tls.sh | :heavy_check_mark: |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |

File: cmd/dummy-handlers.go [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+	// Validate if bucket exists, before proceeding further...
+	_, err := objAPI.GetBucketInfo(ctx, bucket)
+	if err != nil {
+		writeErrorResponse(w, toAPIErrorCode(ctx, err), r.URL, guessIsBrowserReq(r))
+		return
+	}
+
+	tags := &tagging{}
+	tags.TagSet.Tag = append(tags.TagSet.Tag, tagElem{
+		Key:   "",
```
Comment (kannappanr) [inline cmd/dummy-handlers.go]: This tagElem need not be added at all as both the Key and Value are empty .

File: cmd/dummy-handlers.go [hunk 2]
```diff
@@ -0,0 +1,204 @@
...
+
+	// Validate if bucket exists, before proceeding further...
+	_, err := objAPI.GetBucketInfo(ctx, bucket)
+	if err != nil {
+		writeErrorResponse(w, toAPIErrorCode(ctx, err), r.URL, guessIsBrowserReq(r))
+		return
+	}
+
+	tags := &tagging{}
+	tags.TagSet.Tag = append(tags.TagSet.Tag, tagElem{
```
Comment (kannappanr) [inline cmd/dummy-handlers.go]: This tagElem need not be added at all as both the Key and Value are empty .

File: cmd/dummy-handlers.go [hunk 3]
```diff
@@ -0,0 +1,204 @@
...
+	// Validate if object exists, before proceeding further...
+	_, err := objAPI.GetObjectInfo(ctx, bucket, object, ObjectOptions{})
+	if err != nil {
+		writeErrorResponse(w, toAPIErrorCode(ctx, err), r.URL, guessIsBrowserReq(r))
+		return
+	}
+
+	tags := &tagging{}
+	tags.TagSet.Tag = append(tags.TagSet.Tag, tagElem{
+		Key:   "",
```
Comment (kannappanr) [inline cmd/dummy-handlers.go]: This tagElem need not be added at all as both the Key and Value are empty .

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :x: [more...](#aaab1968-21e4-47a7-881e-f859ff03b033) |
| mint-worm.sh | :x: [more...](#82715f20-aed8-474a-9bbd-98e853d140da) |
| mint-gateway-nas.sh | :x: [more...](#22434e48-ac81-445f-b822-92e6fe48be03) |
<a name="82715f20-aed8-474a-9bbd-98e853d140da"></a>
### 7040-12c3be4/mint-worm.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.61:30642
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            worm
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp e8801b7c04a0:/mint/log /tmp/mint-logs'

(1/1) Running worm tests ... FAILED in 0 seconds
{
  "alert": "",
  "args": {
    "bucketName": "bucket-zrzhj52gjlmv9e8pyont9b76fw9t3zc43lhvcgla200jvlfrn1l7gtx",
    "expiry": 60000000000,
    "objectName": "testObject"
  },
  "duration": 284,
  "error": "RequestError: send request failed\ncaused by: Put http://72.28.97.61:30642/bucket-zrzhj52gjlmv9e8pyont9b76fw9t3zc43lhvcgla200jvlfrn1l7gtx: dial tcp 72.28.97.61:30642: connect: connection refused",
  "function": "PutAndDelete",
  "message": "WORM_MODE ON  - CreateBucket Failed",
  "name": "test worm mode",
  "status": "FAIL"
}

Executed 0 out of 1 tests successfully.
```
<a name="aaab1968-21e4-47a7-881e-f859ff03b033"></a>
### 7040-12c3be4/mint-large-bucket.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.56:31447
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp 92656d4c3ae6:/mint/log /tmp/mint-logs'

(1/13) Running aws-sdk-go tests ... done in 1 seconds
(2/13) Running aws-sdk-java tests ... done in 1 seconds
(3/13) Running aws-sdk-php tests ... done in 48 seconds
(4/13) Running aws-sdk-ruby tests ... done in 13 seconds
(5/13) Running awscli tests ... FAILED in 10 seconds
{
  "name": "awscli",
  "duration": 3508,
  "function": "aws --endpoint-url http://72.28.97.56:31447 s3api head-object --bucket awscli-mint-test-bucket-20630 --key prefix/directory/\n",
  "status": "FAIL",
  "error": "An error occurred (404) when calling the HeadObject operation: Not Found"
}

Executed 4 out of 13 tests successfully.
```
<a name="22434e48-ac81-445f-b822-92e6fe48be03"></a>
### 7040-12c3be4/mint-gateway-nas.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.58:31295
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp e5ceb55155e5:/mint/log /tmp/mint-logs'

(1/13) Running aws-sdk-go tests ... done in 0 seconds
(2/13) Running aws-sdk-java tests ... done in 0 seconds
(3/13) Running aws-sdk-php tests ... done in 41 seconds
(4/13) Running aws-sdk-ruby tests ... FAILED in 2 seconds
{
  "name": "aws-sdk-ruby",
  "function": "presignedPut(bucket_name,file_name)",
  "args": {
    "bucket_name": "aws-sdk-ruby-bucket-f78c89da0848",
    "file_name": "datafile-1-MB"
  },
  "duration": 21.15,
  "error": "Connection reset by peer",
  "status": "FAIL"
}

Executed 3 out of 13 tests successfully.
```

File: cmd/dummy-handlers.go (hunk 1 above)
Comment (ebozduman) [reply cmd/dummy-handlers.go]: OK

File: cmd/dummy-handlers.go (hunk 2 above)
Comment (ebozduman) [reply cmd/dummy-handlers.go]: OK

File: cmd/dummy-handlers.go (hunk 3 above)
Comment (ebozduman) [reply cmd/dummy-handlers.go]: OK

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :x: [more...](#d412b454-dae4-4d7c-a1fb-1cee2f634315) |
<a name="d412b454-dae4-4d7c-a1fb-1cee2f634315"></a>
### 7040-b76b694/mint-gateway-nas.sh.log:
```
Running with
SERVER_ENDPOINT:      72.28.97.52:30304
ACCESS_KEY:           minio
SECRET_KEY:           ***REDACTED***
ENABLE_HTTPS:         0
SERVER_REGION:        us-east-1
MINT_DATA_DIR:        /mint/data
MINT_MODE:            full
ENABLE_VIRTUAL_STYLE: 0

To get logs, run 'docker cp d6211f307692:/mint/log /tmp/mint-logs'

(1/13) Running aws-sdk-go tests ... done in 0 seconds
(2/13) Running aws-sdk-java tests ... done in 1 seconds
(3/13) Running aws-sdk-php tests ... done in 41 seconds
(4/13) Running aws-sdk-ruby tests ... FAILED in 1 seconds
{
  "name": "aws-sdk-ruby",
  "function": "presignedPut(bucket_name,file_name)",
  "args": {
    "bucket_name": "aws-sdk-ruby-bucket-4489db11e41e",
    "file_name": "datafile-1-MB"
  },
  "duration": 20.23,
  "error": "Connection reset by peer",
  "status": "FAIL"
}

Executed 3 out of 13 tests successfully.
```

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |

**Review by kannappanr — APPROVED** [review]
LGTM. Tested most of the functions with AWS Cli

**Comment by minio-ops** [comment]
## Mint Automation
| Test | Result |
|:--- |:--- |
| mint-compression-xl.sh | :heavy_check_mark: |
| mint-xl.sh | :heavy_check_mark: |
| mint-large-bucket.sh | :heavy_check_mark: |
| mint-compression-dist-xl.sh | :heavy_check_mark: |
| mint-compression-fs.sh | :heavy_check_mark: |
| mint-worm.sh | :heavy_check_mark: |
| mint-fs.sh | :heavy_check_mark: |
| mint-dist-xl.sh | :heavy_check_mark: |
| mint-gateway-nas.sh | :heavy_check_mark: |

**Review by poornas — APPROVED** [review]
LGTM

**Pull request merged**